src/index.js

```diff
@@ -1,167 +1,249 @@
 import Request from 'backendless-request'
+import Utils from './utils'
+
+const DEFAULT_PROPS = {
+  applicationId : null,
+  secretKey     : null,
+  serverURL     : 'https://api.backendless.com',
+  debugMode     : true,
+  ServerCode    : null,
+  XMLHttpRequest: typeof XMLHttpRequest !== 'undefined' ? XMLHttpRequest : undefined,
+}
 
-import LocalVars from './local-vars'
-
-const root = (typeof self === 'object' && self.self === self && self) ||
+const root = (
+  (typeof self === 'object' && self.self === self && self) ||
   (typeof global === 'object' && global.global === global && global)
+)
 
 const previousBackendless = root && root.Backendless
 
-const Backendless = {
+// Backendless supports two signatures for the initApp method
+// two args - applicationId {String} and secretKey {String}
+// or one argument - whole set of options {Object}
+const parseArgs = (...args) => {
+  const [applicationId, secretKey] = args
 
-  get debugMode() {
-    return LocalVars.debugMode
-  },
+  if (Utils.isObject(applicationId)) {
+    return applicationId
+  }
 
-  set debugMode(debugMode) {
-    LocalVars.debugMode = !!debugMode
+  return {
+    applicationId,
+    secretKey
+  }
+}
 
-    require('./rt').setRTDebugMode(LocalVars.debugMode)
-  },
+class Backendless {
+  constructor(props) {
+    this.initProps(props)
 
-  get serverURL() {
-    return LocalVars.serverURL
-  },
+    this.Request = Request
+  }
 
-  set serverURL(serverURL) {
-    LocalVars.serverURL = serverURL
-  },
+  /**
+   * @param {Object} props
+   */
+  initProps(props) {
+    for (const key in DEFAULT_PROPS) {
+      if (DEFAULT_PROPS.hasOwnProperty(key) && props.hasOwnProperty(key)) {
+        const privateKey = `__${ key }`
 
-  get XMLHttpRequest() {
-    return LocalVars.XMLHttpRequest
-  },
+        this[privateKey] = props[key]
+      }
+    }
+  }
 
-  set XMLHttpRequest(XMLHttpRequest) {
-    LocalVars.XMLHttpRequest = XMLHttpRequest
-  },
+  /**
+   * @param {string|Object} applicationId|options
+   * @param {string} [secretKey]
+   * @returns {Backendless}
+   */
+  initApp(...args) {
+    const { standalone, ...restProps } = parseArgs(...args)
```

Comment:
Where `standalone` comes from?

Reply:
it is an option that customer will set to get a separate backendless app client